backends/jms/src/main/java/org/hibernate/search/backend/impl/jms/AbstractJMSHibernateSearchController.java

```diff
@@ -69,8 +70,16 @@ public void onMessage(Message message) {
 				log.messageReceivedForUndefinedIndex( indexName );
 				return;
 			}
-			queue = indexManager.getSerializer().toLuceneWorks( (byte[]) objectMessage.getObject() );
-			indexManager.performOperations( queue, null );
+
+			LuceneWorkSerializer luceneWorkSerializer = integrator.getServiceManager().requestService( LuceneWorkSerializer.class );
```

Comment:
One of my concerns is that this is a change that inpacts several integration points. I'm aware we're ok to change SPIs, but not all SPI are equal.. 

But I see your point in wanting to decouple it. Let's say we agree on moving the reference to `LuceneWorkSerializer`, I think what we need is:

    integrator.getLuceneWorkSerializer();

I would still keep the IndexManager#getSerializer() method around to keep such users happy, and deprecate it. (It would be implemented by simply delegating to  `integrator.getLuceneWorkSerializer()`